### PR TITLE
refactor(runtime): make ForwardContext tensor-free (restore "Do not contain Tensor")'

### PR DESCRIPTION
**Pull request by rjzhb**
## Goal

`ForwardContext` documents itself as **"Do not contain Tensor"**, yet five loose `torch.Tensor` fields had crept in over time: `req_to_page`, `gather_ids`, `draft_seq_lens_buf`, `accept_lengths`, and `dsa_swa_slot_mapping`.

This PR makes the context tensor-free again—each tensor now travels as an **explicit forward argument**, lives in a **persistent buffer**, or is folded into the **attention metadata**, so nothing loose remains on the per-forward context.

This is not a bug fix: these tensors were only read on eager paths or were already persistent buffers, so CUDA graph replay is unaffected. This is invariant/hygiene cleanup—keeping the context from silently becoming a tensor grab bag.

## Status — all five removed

* [x] `draft_seq_lens_buf` — dropped; draft cache lengths reuse the drafter's `seq_lens` buffer, passed explicitly
* [x] `accept_lengths` — passed explicitly as a forward kwarg (drafter first step only)
* [x] `req_to_page` — moved to a persistent page-table buffer; all reads now use explicit arguments
* [x] `gather_ids` — computed per forward pass in the executor (`_target_gather_ids`) and passed explicitly to the model and logits
* [x] `dsa_swa_slot_mapping` — folded into DeepSeek-V4 attention metadata (set and cleared within a single forward pass)

`ForwardContext` now carries no `torch.Tensor` fields—see `context.py:37`.

### CODE REVIEW

**Review by nperrin-fr — APPROVED** [review]
lgtm

File: python/tokenspeed/runtime/models/extensible.py [hunk 1]
```diff
@@ -189,8 +210,13 @@ def forward(
...
-            input_ids, positions, ctx, out_hidden_states
-        )
+        output_args = (input_ids, positions, ctx, out_hidden_states)
+        if self._output_processor_accepts_gather_ids:
+            logits_output: LogitsProcessorOutput = self.output_processor(
+                *output_args, gather_ids=gather_ids
+            )
+        else:
+            # Preserve the four-argument contract for existing custom processors.
+            logits_output = self.output_processor(*output_args)
```
[the remark this answers is not in the document]
Comment (rjzhb) [reply python/tokenspeed/runtime/models/extensible.py:220]: gather_ids only matters on prefill; decode-only processors correctly get None. The repo's only processor (the base class) already takes gather_ids, and any out-of-tree processor running prefill must now accept it explicitly — that's the intended API change, not a gap.
